Add sample versions of standard deviation and variance functions

bpintea opened the pull request:
This PR adds the sampling variations of the standard deviation and variance agg functions, `STDDEV_SAMP` and `VAR_SAMP` respectively.

Closes #57950

bpintea committed:
Add STDDEV_SAMP, VAR_SAMP

This commit adds the sampling variations of the standard deviation and
variance agg functions.

bpintea committed:
test suites fixes

Correct CSV tests.

elasticmachine commented:
Pinging @elastic/es-ql (:Query Languages/SQL)

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into enh/add_sample_stddev_variance

matriv commented on x-pack/plugin/sql/qa/server/src/main/resources/docs/docs.csv-spec:
```diff
@@ -1615,6 +1637,28 @@ SELECT MIN(salary / 24.0) AS min, MAX(salary / 24.0) AS max, VAR_POP(salary / 24
+       min        |       max        |     varsamp
+------------------+------------------+----------------
+1055.1666666666667|3124.9583333333335|332278.830154847
+
```
extra empty line

matriv reviewed (commented):
Looks good, but can we have some unit tests in QueryTranslatorTests?
It would be great to show that when having those aggs on the same field we use ExtendedStats agg)

costin reviewed (approved):
LGTM

astefan reviewed (approved):
LGTM
Agree with @matriv: a query translator test would make this feature complete.

bpintea committed:
Address review comments

- add QueryTranslatorTests for STDDEV* and VAR* functions;
- remove superfluous empty line.

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into enh/add_sample_stddev_variance

matriv reviewed (approved):
LGTM, thx!
